api/controllers/v0.3/DependentAccountAPI.js

```diff
@@ -5,8 +5,57 @@ const IlsClient = require("./IlsClient");
 /**
  *
  */
-const DependentEligibilityAPI = (args) => {
+const DependentAccountAPI = (args) => {
   const ilsClient = args["ilsClient"];
+  let parentPatronData;
+
+  /**
+   * isPatronEligible(barcode)
+   * The main function of this class. It gets a patron data object from the ILS
+   * and it verifies it has the correct p-type. If it does, then it checks if
+   * it can create another dependent account. It returns a response stating
+   * whether the patron is eligible or not and a description if they are not.
+   *
+   * @param {string} barcode
+   */
+  const isPatronEligible = async (barcode) => {
+    if (!ilsClient) {
+      throw new NoILSClient(
+        "ILS Client not set in the Dependent Eligibility API."
+      );
+    }
+
+    let response = {};
+    const patron = await getPatronFromILS(barcode);
+    // Set the fetched patron data object into the global variable so
+    // it can be accessed by `getPatron`. This is specifically set here and not
+    // in `getPatronFromILS` because we want to make sure that the eligibility
+    // check was ran in order to retrieve a patron.
+    parentPatronData = patron;
+    // First, check that they have a valid ptype to be able to create
+    // dependent accounts.
+    const hasValidPtype = checkPType(patron.patronType);
+
+    if (hasValidPtype) {
+      // Great, they have a valid ptype. Now check that they have not reached
+      // the dependent account limit.
+      const canCreateDependents = checkDependentLimit(patron.varFields);
```

Comment:
It might be helpful to specify in the comment what the limit is (unless we think it's about to change)

Reply:
Yeah, I wrote that the limit is three in the specific function that does that, but can include it here.